ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -355,6 +368,18 @@ extern "C" void *__ykllvmwrap_irtrace_compile(char *FuncNames[], size_t BBs[],
     }
   }
   Builder.CreateRetVoid();
+
+  // Fix initialisers/referrers for copied global variables.
+  // FIXME Do we also need to copy Linkage, MetaData, Comdat?
+  for (GlobalVariable *G : cloned_globals) {
+    GlobalVariable *NewGV = cast<GlobalVariable>(VMap[G]);
+    if (G->isDeclaration())
```

Comment:
What's this line about?

Reply:
Not entirely sure (this is mostly copied from LLVM's `cloneModule` function). But I assume that if it's a declaration you don't need to initialise it.

Reply:
Ah. I vaguely remember being confused by "define" vs "declare" when I was working in rustc. @bjorn3's comment reminded me.

We should double check that we understand what's going on there.

Reply:
5d22c88 should address this. There was no changes needed, as we already make sure that any operand/instruction only gets cloned once into the `VMap`.